invenio_accounts/alembic/dfbdf43a3e96_separate_login_info.py

```diff
@@ -0,0 +1,112 @@
+
+
+def upgrade():
+    """Upgrade database."""
+    # create the new login information table
+    op.create_table(
+        "accounts_user_login_information",
+        sa.Column("user_id", sa.Integer(), nullable=False),
+        sa.Column("current_login_at", sa.DateTime(), nullable=True),
+        sa.Column("current_login_ip", IPAddressType(), nullable=True),
+        sa.Column("last_login_at", sa.DateTime(), nullable=True),
+        sa.Column("last_login_ip", IPAddressType(), nullable=True),
+        sa.Column("login_count", sa.Integer(), nullable=True),
+        sa.ForeignKeyConstraint(
+            ["user_id"],
+            ["accounts_user.id"],
+            name="fk_accounts_login_information_user_id",
+        ),
+        sa.PrimaryKeyConstraint(
+            "user_id", name=op.f("pk_accounts_user_login_information")
+        ),
+    )
+
+    # move information from the user table into the new table
+    op.execute(
+        "INSERT INTO accounts_user_login_information "
+        "(user_id, current_login_at, current_login_ip, last_login_at, last_login_ip, login_count) "  # noqa
+        "SELECT id, current_login_at, current_login_ip, last_login_at, last_login_ip, login_count "  # noqa
+        "FROM accounts_user;"
+    )
+
+    # remove columns from the user table
+    op.drop_column("accounts_user", "login_count")
+    op.drop_column("accounts_user", "last_login_at")
+    op.drop_column("accounts_user", "current_login_ip")
+    op.drop_column("accounts_user", "last_login_ip")
+    op.drop_column("accounts_user", "current_login_at")
+
+
+def downgrade():
+    """Downgrade database."""
+    # create the old columns on the user table
+    op.add_column(
+        "accounts_user",
+        sa.Column("current_login_at", sa.DateTime(), nullable=True),
+    )
+    op.add_column(
+        "accounts_user",
+        sa.Column("current_login_ip", IPAddressType(), nullable=True),
+    )
+    op.add_column(
+        "accounts_user",
+        sa.Column("last_login_at", sa.DateTime(), nullable=True),
+    )
+    op.add_column(
+        "accounts_user",
+        sa.Column("last_login_ip", IPAddressType(), nullable=True),
+    )
+    op.add_column(
+        "accounts_user",
+        sa.Column("login_count", sa.Integer(), nullable=True),
+    )
+
+    # move the data from the new table to the old columns
+    dialect_name = op._proxy.migration_context.dialect.name
+    if dialect_name == "postgresql":
+        op.execute(
+            "UPDATE accounts_user SET "
+            " current_login_at = li.current_login_at, "
+            " current_login_ip = li.current_login_ip, "
+            " last_login_at = li.last_login_at, "
+            " last_login_ip = li.last_login_ip, "
+            " login_count = li.login_count "
+            "FROM accounts_user_login_information AS li "
+            "WHERE accounts_user.id = li.user_id;"
+        )
+    elif dialect_name == "mysql":
+        # TODO untested; maybe it would be possible with sqlalchemy core?
```

Comment:
:eyes:

Reply:
no big deal I think......downgrading rarely ever happens

Reply:
i dropped the untested mysql query for data retention on downgrade, as this isn't expected to be a common use case, and the information isn't precisely vital